### DISCUSSION TITLE
Connecting multiple devices to one Waveshare RS485 to ETH server

### DISCUSSION THREAD

**Discussion by tomasko25** [wills106/homeassistant-solax-modbus#308] [Q&A] [answered]
Hi,

I'd like to monitor my Solax inverter and energy meter (which is not connected with inverter) in my Home Assistant. Both devices have Modbus output so I was thinking about getting 2 of these converters https://www.waveshare.com/RS485-TO-ETH-B.htm. But - would it be possible to connect all into one converter (for exapmle in "daisy-chain" topology) and be able to read data from both devices separately in Home Assistant? 

I am not expert on ModBus protocol so I would appreciate your answers or ideas.

Thanks

**Comment by wills106** [comment, marked as answer]
It should be possible to have multiple devices on the same adaptor as long as the addressing is unique.
The problem you will probably find though is the Inverter will want to communicate at one baud rate and the Modbus Power Meter at a different rate. Unless the Power Meter allows you to change the baud to match the Inverter?

**Comment by alienatedsec** [comment]
@tomasko25 I also use multiple Waveshare devices. Have a look at the below diagram and its description available [here](https://github.com/fboundy/ha_solis_modbus/issues/4#issuecomment-1436737744)

![image](https://user-images.githubusercontent.com/73167064/218755107-58353acb-e95c-4bda-93c2-a34a28608a0c.png)

If you want to use only one gateway (I believe you call it a converter) you need to make sure that both, the Power Meter, and the inverter have different slave IDs. 

There are plenty of examples, but the one below seems to be the best illustration of a similar configuration. It is achievable with Waveshare devices too.

![image](https://user-images.githubusercontent.com/73167064/222578042-ee42b779-9109-4fc5-8f75-0ea94f049366.png)
